designs/2018-processors-improvements/README.md

```diff
@@ -0,0 +1,271 @@
+    ]
+};
+```
+
+In this configuration, the named processor `markdown` from the `eslint-plugin-markdown` plugin is applied only to files with the extension `*.md`. This mimics the behavior of the extension-named processor that `eslint-plugin-markdown` currently exports.
+
+The `CLIEngine` method `getConfigForFile()` would automatically get the correct value for `processors` when the configuration is calculated.
+
+**Note:** End-users would still need to provide `--ext .md` on the command line for ESLint to automatically lint files ending with `.md`. No part of this proposal changes the behavior of `--ext`.
+
+### Apply Multiple Processors
+
+In special cases, end-users may want to apply multiple processors to files. This can be accomplished by listing multiple named processors using the `processors` configuration option. For example:
+
+```js
+module.exports = {
+    plugins: ["markdown", "vue"],
+    parserOptions: {
+        ecmaVersion: 2018
+    },
+    overrides: [
+        {
+            files: ["*.md"],
+            processors: ["markdown/markdown", "vue/vue"]
+        }
+    ]
+};
+```
+
+This configuration applies two named processors to all files matching `*.md`. First, the `markdown` named processor from `eslint-plugin-markdown` is applied. The result of that preprocessing is then passed to the `vue` named processor from `eslint-plugin-vue`. The result of that preprocessing is then passed to ESLint for linting. When ESLint has completed linting, the reverse sequence occurs: the linting results are first passed to the `vue` named processor from `eslint-plugin-vue`, and once that postprocessing is complete, the result is then passed to the `markdown` named processor from  `eslint-plugin-markdown`.
+
+End-users can also mix and match processors, such as in this example:
+
+```js
+module.exports = {
+    plugins: ["markdown", "vue", "html"],
+    parserOptions: {
+        ecmaVersion: 2018
+    },
+    overrides: [
+        {
+            files: ["*.md"],
+            processors: ["markdown/markdown", "vue/vue"]
+        },
+        {
+            files: ["*.htm", "*.html"],
+            processors: ["html/html", "vue/vue"]
+        }
+    ]
+};
+```
+
+This configuration uses the same `vue` named processor from `eslint-plugin-vue` for both HTML and Markdown files, but only after an appropriate named processor is used to extract the Vue code.
+
+**Note:** It would be update to the Vue processor to understand the difference between different types of code it may be passed.
+
+### Virtual Filenames for Code Blocks
+
+The `preprocess()` method inside of a processor can now return an array containing either strings (the current behavior) or an object with two properties:
+
+* `filePath` - (string) a virtual file path to map the code block to
+* `text` - (string) the source code text
+
+For example:
+
+```js
+module.exports = {
+    preprocess(text, filename) {
+        
+        const {vueText, jsText } = doSomething(text, filename);
```

Comment:
How does the processor determine whether it should return `vueText` as well?

In general, ESLint can handle only JavaScript, so processor should return only JavaScript blocks.
For example, maybe `eslint-plugin-markdown` needs to have `markdown/js`, `markdown/js,html`, `markdown/js,ts`, `markdown/js,vue`, `markdown/js,html,ts`, `markdown/js,html,vue`, `markdown/js,ts,vue`, and `markdown/js,html,ts,vue` if there is no configuration way.

Reply:
That's a good point. I was trying to make up a useful example but I think I confused the point. I will fix this.